packages/smooth_app/lib/pages/product/common/product_query_page.dart

```diff
@@ -410,34 +411,18 @@ class _ProductQueryPageState extends State<ProductQueryPage>
         ),
       );
 
-  _Action _getRefreshAction(
-    final AppLocalizations appLocalizations,
-  ) =>
-      _Action(
-        text: appLocalizations.label_refresh,
-        iconData: Icons.refresh,
-        onPressed: () async {
-          final bool? success =
-              await _loadAndRefreshDisplay(_model.loadFromTop());
-          if (success == true) {
-            _scrollToTop(instant: true);
-          }
-        },
-      );
-
   void retryConnection() =>
       setState(() => _model = _getModel(widget.productListSupplier));
 
   ProductQueryModel _getModel(final ProductListSupplier supplier) =>
       ProductQueryModel(supplier);
 
   Future<void> refreshList() async {
-    final ProductListSupplier? refreshSupplier =
-        widget.productListSupplier.getRefreshSupplier();
-    setState(
-      // How do we refresh a supplier that has no refresher? With itself.
-      () => _model = _getModel(refreshSupplier ?? widget.productListSupplier),
-    );
+    final bool success = await _model.loadFromTop();
```

Comment:
A `try`/`catch` would not hurt. Cf. `_downloadNextPage`.

Reply:
so, should we show `snackbar` in catch block?

Reply:
What snackbar? There's not snackbar in your code, is there?
As long as you check `if(mounted)` after an `await`, I guess calling a snackbar is OK.

Reply:
I haven't added snackbar, (could create another issue for it). I mentioned it to handle `catch` block, and to let user know?